### PR TITLE
Test reserved state read and write through RawRepsitory

### PR DESCRIPTION
**Pull request by junha1**
This PR also includes various setup for integration tests (will be covered in the next PR)

### CODE REVIEW

File: repository/src/raw/implementation.rs [hunk 1]
```diff
@@ -294,39 +294,16 @@ impl RawRepositoryImplInner {
...
-                    fs::create_dir(path.clone()).unwrap();
-                }
-                fs::write(path.join(Path::new("genesis_info.json")), genesis_info).unwrap();
-                fs::write(
-                    path.join(Path::new("consensus_leader_order.json")),
-                    consensus_leader_order,
-                )
-                .unwrap();
-                fs::write(path.join(Path::new("version")), version).unwrap();
+                let path = self.repo.workdir().unwrap().to_str().unwrap();
```
Comment (JeongHunP) [inline repository/src/raw/implementation.rs:297]:
```suggestion
                let path = self.repo.workdir()?.to_str().unwrap();
```

File: repository/src/raw/implementation.rs [hunk 2]
```diff
@@ -601,37 +578,11 @@ impl RawRepositoryImplInner {
 
     pub(crate) fn read_reserved_state(&self) -> Result<ReservedState, Error> {
         let path = self.repo.workdir().unwrap().to_str().unwrap();
```
Comment (JeongHunP) [inline repository/src/raw/implementation.rs:580]:
```suggestion
        let path = self.repo.workdir()?.to_str().unwrap();
```

File: repository/src/raw/tests.rs [hunk 3]
```diff
@@ -448,3 +450,25 @@ async fn remote() {
...
+
+#[tokio::test]
+async fn reserved_state() {
+    let td = TempDir::new().unwrap();
+    let path = td.path();
+    let mut repo = init_repository_with_initial_commit(path).await.unwrap();
+
+    let (rs, _) = simperby_test_suite::generate_standard_genesis(10);
+
+    repo.checkout(MAIN.into()).await.unwrap();
```
Comment (JeongHunP) [inline repository/src/raw/tests.rs:462]: repo currently is in "main" branch, so I don't think this line is needed.

File: repository/src/raw/tests.rs [hunk 4]
```diff
@@ -448,3 +450,25 @@ async fn remote() {
...
+    repo.create_semantic_commit(SemanticCommit {
+        title: "test".to_owned(),
+        body: "test-body".to_owned(),
+        diff: Diff::Reserved(Box::new(rs.clone())),
+    })
+    .await
+    .unwrap();
+    repo.checkout(MAIN.into()).await.unwrap();
```
Comment (JeongHunP) [inline repository/src/raw/tests.rs:470]: After making a commit, the branch is automatically changed to point at the new commit. 
So this line might not be needed, too.
But please check after removing this.

File: repository/src/raw/tests.rs (hunk 3 above)
Comment (junha1) [reply repository/src/raw/tests.rs:462]: that is true but its not trivial. (it's highly dependent on the implementation of `init_repository_with_initial_commit`).

File: repository/src/raw/tests.rs (hunk 4 above)
Comment (junha1) [reply repository/src/raw/tests.rs:470]: ok. fixed

File: repository/src/raw/tests.rs (hunk 3 above)
Comment (JeongHunP) [reply repository/src/raw/tests.rs:462]: Agree

File: common/tests/integration_tests.rs [hunk 5]
```diff
@@ -1,55 +1,14 @@
 use light_client::*;
 use merkle_tree::*;
-use reserved::ReservedState;
 use simperby_common::{verify::CommitSequenceVerifier, *};
+use simperby_test_suite::*;
 
 #[test]
 fn basic1() {
     let n = 10;
```
Comment (TomTaehoonKim) [inline common/tests/integration_tests.rs:8]:
```suggestion
    let member_number = 10;
```
or just 10 at line 9

I had to look up the declaration of `generate_standard_genesis` to find the meaning of `n`

File: repository/src/raw/reserved_state.rs [hunk 6]
```diff
@@ -86,49 +86,7 @@ mod tests {
...
-        };
-        let members: Vec<Member> = vec![member1, member2];
-
-        let reserved_state = ReservedState {
-            genesis_info,
-            members,
-            consensus_leader_order: vec![0, 1],
-            version: "0.1.0".to_string(),
-        };
+        let (reserved_state, _) = simperby_test_suite::generate_standard_genesis(10);
```
Comment (TomTaehoonKim) [inline repository/src/raw/reserved_state.rs:92]: some `rs`, some `reserved_state`

not necessary but I think consistent naming is good

File: test-suite/src/lib.rs [hunk 7]
```diff
@@ -1 +1,90 @@
+use simperby_node::simperby_common::*;
 
+#[cfg(target_os = "windows")]
```
Comment (TomTaehoonKim) [inline test-suite/src/lib.rs:3]: do we need some test added for Mac?

File: test-suite/src/lib.rs [hunk 8]
```diff
@@ -1 +1,90 @@
...
+    let mut child = tokio::process::Command::new("sh")
+        .arg("-c")
+        .arg(command.as_ref())
+        .spawn()
+        .expect("failed to execute process");
+    let ecode = child.wait().await.expect("failed to wait on child");
+    assert!(ecode.success());
+}
+
+/// Generates a standard test chain config returning the genesis reserved-state.
```
Comment (TomTaehoonKim) [inline test-suite/src/lib.rs:26]: and the key pairs

File: repository/src/raw/tests.rs [hunk 9]
```diff
@@ -448,3 +450,24 @@ async fn remote() {
...
+
+#[tokio::test]
+async fn reserved_state() {
+    let td = TempDir::new().unwrap();
+    let path = td.path();
+    let mut repo = init_repository_with_initial_commit(path).await.unwrap();
+
+    let (rs, _) = simperby_test_suite::generate_standard_genesis(10);
+
+    repo.checkout(MAIN.into()).await.unwrap();
```
Comment (TomTaehoonKim) [inline repository/src/raw/tests.rs:462]: do we have main branch? what is it?

Comment (junha1) [reply repository/src/raw/tests.rs:462]: Please note that this is a test for RawRepository, not DistributedRepository.

File: repository/src/raw/reserved_state.rs (hunk 6 above)
Comment (junha1) [reply repository/src/raw/reserved_state.rs:92]: They're already interchangeably used in the project. This single PR can't fix all.

File: test-suite/src/lib.rs (hunk 8 above)
Comment (junha1) [reply test-suite/src/lib.rs]: fixed

File: repository/src/raw/reserved_state.rs (hunk 6 above)
Comment (TomTaehoonKim) [reply repository/src/raw/reserved_state.rs:92]: @junha1 let's just do all those kinds of stuffs after the genesis

File: repository/src/raw/tests.rs (hunk 9 above)
Comment (TomTaehoonKim) [reply repository/src/raw/tests.rs:462]: right, understood

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by YoonhoKim0527 — APPROVED** [review]
LGTM

**Pull request merged**